### ISSUE TITLE
Heredoc containing \n cannot be parsed for indenting

### ISSUE THREAD

**Issue by karptonite** [prettier/plugin-php#2015]
[Playground link](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCACsAOj4gWMCG3iyVAYXQBbH2wAFAAkA-BdBQxAOVwBjKABlGABPNWswbAgAWgAmJxBoAEdBeGgA12V-PTkFEGIRCBgytgq2KoMAXxYoCngYdBQ0eCxJUDZsGDaRTtrMMEEJvBAAHgB+YzAAHUwpQUwYjEwAAil0dAAKAEpD4C3Du8OAEh8obGxmqEO7AF5Dpb+AUWQolu9wCajGH2gbAohxo8Gwh2er3edg2INIHzYYHQsnQyngAC8PicILDqB9iDEsAA3KCYRB7KCoo73QGiADcW3aWxA7SAA)

Input:
```php
<?php
function foo() {
    $message .= <<<EOT
    Please read this message.\n  We apologize for the inconvenience.
    EOT;
}
```

Output:
```
Invalid body indentation level (expecting an indentation at least 4) on line 5

  3 |     $message .= <<<EOT
  4 |     Please read this message.\n  We apologize for the inconvenience.
> 5 |     EOT;
    |    ^
  6 | }
  7 |
```

Expected:
```php
<?php
function foo() {
    $message .= <<<EOT
    Please read this message.\n  We apologize for the inconvenience.
    EOT;
}
```

Basically, the `\n` in the middle of the heredoc appears to confuse either the parser or prettier.

**Comment by cseufert** [comment]
This is definitely an issue in the parser, it's seeing the `\n` in the middle of the string as a newline when calculating the indent level of the heredoc.

I am not sure how to fix this one, it seems the \n is being turned into a new line, but I am not sure how it keeps that value independent from the actual newline.

**Comment by claytonrcarter** [comment]
I just ran into this, too, and was able to reduce it to `\n` followed by a character. If the `\n` is the last char on the line, it's not an issue.
```php
<?php

if (true) {
    echo <<<STR
    \na
    STR;
}
```

```sh
❯ node_modules/.bin/prettier --check foo.php
Checking formatting...
foo.php
[error] foo.php: SyntaxError: Invalid body indentation level (expecting an indentation at least 4) on line 6
[error]   4 |     echo <<<STR
[error]   5 |     \na
[error] > 6 |     STR;
[error]     |    ^
[error]   7 | }
[error]   8 |
All matched files use Prettier code style!

❯ node_modules/.bin/prettier --version
3.2.5

❯ npm list | grep plugin-php
├── @prettier/plugin-php@0.22.2
```

The original string that caused this for us was an SQL statement that involved the line `LINES TERMINATED BY '\n'`
